mrinoybanerjee/parrot-ai
# Enhanced evaluation pipeline

# Enhanced Evaluation Script for Parrot-AI

## Description
This PR introduces significant improvements to the evaluation script for Parrot-AI, providing a more comprehensive and insightful analysis of the system's performance.

## Changes
1. Added hardware configuration reporting
2. Expanded API endpoint testing (root, generate_conversation, generate_summary)
3. Implemented response quality metrics (cosine similarity and Levenshtein distance)
4. Enhanced visualizations for latency, response time, and quality metrics
5. Improved markdown output with detailed explanations and interpretations

## New Features
- Hardware info gathering using `psutil`
- Cosine similarity calculation using BERT embeddings
- Levenshtein distance computation for string comparison
- Generation of result tables using `tabulate`
- Creation of three new visualizations:
  - Latency distribution
  - Response time vs request number
  - Similarity and Levenshtein score distributions

## Updated Dependencies
- Added `psutil` for hardware information
- Added `tabulate` for formatting result tables

## How to Test
1. Ensure the backend server is running at `http://localhost:8000`
2. Install updated dependencies: `pip install -r requirements.txt`
3. Run the evaluation script: `python evaluate.py`
4. Check the generated `EVALUATION_RESULTS.md` file and `assets` folder for results and visualizations

## Expected Outcome
- A comprehensive `EVALUATION_RESULTS.md` file containing:
  - Hardware configuration
  - Performance metrics explanations
  - Latency results in tabular format
  - Response quality metrics
  - Links to and interpretations of visualizations
- Three new visualization images in the `assets` folder

## Notes
- This update significantly increases the depth of the evaluation, providing more actionable insights into the system's performance and response quality.
- The script is now more resilient to potential errors and provides clearer logging.
